### ISSUE TITLE
Might want to let users know the ,, actions are not the default.

### ISSUE THREAD

**Issue by laternser** [vimoutliner/vimoutliner#179]
Please make a prominent note that the vimoutliner keystrokes (the default) have changed; and that
the file structure has changed.   I wasted A GREAT DEAL of time.

My advice to long term vim users is to start with a fresh and empty .vim

============
The old ,, strokes are ingrained in my brain.  I just assumed vimoutliner was broken
as vim was behaving oddly.    ... Many odd behaviours that probably relate to other plug-ins.
 
(On Windows I still use vim 7.3 and outliner from 2010.)

After I was forced out of Ubuntu 16.04 a year ago, vim-outliner ceased functioning correctly.

I have been using vim since about 1992, and vimoutliner since (???) before the turn of the century perhaps.
I recall something about a version 0.1.????.    I now have roughly 1,600 otl files in active use.

I just kept migrating .vim package from system to system on linux.  
   !!!  That no longer works.
Suddenly, with 8.2 (starting at 8.013???)  the whole operation of vim went haywire.

I had installed from zip and then git ... but without creating a whole new .vim 
         ... this did not work.
         ... plus I did not know the keystrokes had changed.

I actually recently changed Linux (.deb back to .rpm to another .rpm) flavors in an attempt to correct vim.
Finally  ...  today erasing my long standing .vim pile of plug-in folders.

And discovered the keystrokes of vim-outliner had changed.

I wonder how many other users gave up?

Many (most) of the vimoutliner help pages on the internet are ancient ... and have the wrong directions.

Again, I suggest:
   1) Pre-existing users to vim 8.2 start with a new .vim structure, following the 8.2 structure guidelines.
       And follow the git instructions to install exactly.
   2) Post prominent notice that the keystrokes (by default) have changed.

Each new release of Vim is fixing some of the problems.   I created a Gvim appimage that now functions
on all the systems I use.

The old vimoutliner keystrokes are so ingrained in me that I cannot describe them without doing them.
But the new \ strokes are superior.   I struggled often getting the ,, fast enough.

I probably owe this plug-in (and Steve Litt) in excess of $10,000 in time saved.   Way in excess.

Vim was my first cross platform application (well ... maybe vi ... since Xenix had that back in 1985.)

Tried to post on the list ... but no login permission was forthcoming.
Twenty years plus of vimoutliner ... I was afraid it was over.
Thank you.

**Comment by mcepl** [comment]
Well, that change happened in https://github.com/vimoutliner/vimoutliner/commit/82348f45a5760f5169216327caceb533ccf8d2be (that's  4 Sep 2014). If you haven’t recovered from that change yet, I think it is too late to save you.

**Comment by noelhenson** [comment]
Laternser,

I have felt your pain for many years now. Certain distros have their own 
ways of doing things. I have never been fond of the issues of others 
(distros) changing and remolding how VO should work. 21 years ago, Steve 
and I created VO. I was the major programmer. I probably have spent over 
2,000 hours on it. Steve was the guiding light on functionality. I 
implemented checkboxes, checkbox percentages, checkbox weighting, 
'concealed' functions, VO math, VO magic, otl2html, otl2ooimpress, 
otl2table, otl2tags, otlgrep, otlreorder, otlsplit, and other otl to 
whatever tools. I worked hard to get the conceal function implemented so 
'math' and other inline functions and syntax could be hidden.

About 11 years ago, I got fed up with others trying to pry this simple, 
elegant outliner tool into their 'way' of implementing things and 
stopped being involved. I still have my own, trusty version of VO that I 
still use to this day. Back in the day, it was a simple shell script 
that installed it all; and it ALWAYS worked.

I apologize for Matej comment about your not being able to be saved. He 
is a good guy and I have rarely ever heard a comment like this from him.

To everyone else in the world, VO is still a great tool.

Noel
Author of VO

**Comment by mcepl** [comment]
> About 11 years ago, I got fed up with others trying to pry this simple, elegant outliner tool into their 'way' of implementing things and stopped being involved. I still have my own, trusty version of VO that I still use to this day.

Hi, Noel,

I am sorry, I have never knew you felt so strongly about this, you never told us (or I was too dim to understand). I thought you were just too busy or you lost your interest, so I was waiting for years that you return and finish #78 (for example).

Actually, this may be the opportunity to give you VO back (if you feel I have stolen it from you and Steve; I really didn't mean it like that). I feel I really need to cut on my upstream work. I am firmly on the side of open source (so no switching to proprietary software; after all, I still work for SUSE), but it is getting really too much. So, I have to get rid of maintaining some projects, which don't have big enough audience and they could be replaced by something else. [rope](https://github.com/python-rope/rope/) fells under this rule, and VO I am afraid as well. I am not sure whether I switch to vimwiki or some vim org-mode replacement, but I think it is the time for me to shed VO. Do you want it? I'll give you ownership of this repo (or to be exact, I will step down from vimoutliner group), and you can finally uncomment your ``let localleader=,,`` again. Do you want it?

Matěj

**Comment by laternser** [comment]
(Readers can skip down to "THE PROBLEM" for a description of what I believe is a VO problem.)

Noel, 

I am pleased you still pay notice to your creation. You have certainly made my life better
for decades now. I also highly appreciate the useabliity of early versions of your work and 
vim-outliner continuing ability to transfer.  Still one of my all time most useful pieces of software.
The simple vim outliner has an personal affection place ranking above CAE packages I use 
that cost thousands.  The only difference being I could probably live without it ... but with sadness. 

My perception is that fewer and fewer people are using this plugin... 
    ... perhaps because of functionality and install problems.

I admit to having a big laugh at my expense when I discovered the default keystroke change ...
which meant having to find the documentation ... which I have not read in 20 years.
Mostly I have been using the old zip files from vim.org.

I was hoping that the plugin could continue another 20 years and more.
Simple, fast, effective, and does not get in the way.  Engineers KISS principle exemplified.

(I always installed manually ... did not even use the shell script ... which sometimes don't work.)
 
VO has been a way of life for me (and plain vim before that ... and vi before that ... but never ed.)
I took no offense at mcepl as it may be true that I am too late to be saved, something that is 
not true of vim outliner.
      I don't doubt for a moment the man hours that were spent creating it.
      I am just glad it was done ... it would not have sprung from my brain.
      The product of thought, effort, and perseverance ... as in all good things.

How I love checkboxes.   Makes the project managers uselessly slow in comparison.
(I have hopes for the future of Everdo... well done package ... almost mouse free
and vim keystroke oriented.)

I probably should start a new topic for the remaining problem of vim outliner.
Been meaning to try and sort it out myself  ... but it would take a large effort ..
and I was unsure if I was the only one experiencing a problem.

Anyway, the changes to Vim (particularly 8.2) have lead to vim ouliner odd behavior.  As far as I can determine
vim outliner has not changed ... vim has.   Nothing simple in the coding of Vim.

I have now tested this with Xbuntu 20.04, Xbuntu 16.04, Xbuntu 18.03, and Oracle (Redhat) linux (current), 
and now openSUSE 15.3.
All combinations demonstrate problems ... and most use vim 8.2.      

I shifted to openSUSE because they still distribute VIM 8.0 and I can use their vim package.
I have compiled vim from source and also have an appimage of v8.2.2825 that I have carried around and tested on 
these distributions.
Early versions of 8.2 were really a mess ... there has been vast improvment over the last year.
(neovim just does not come up to snuff yet.   ... I was really frustrated and gave it a try.)
I love the Gvim keystroke saving menus that allow me to perform some operations left handed with fewer keystrokes.)

Side note: The following probem reproduces faithfully on new installs (new .vimrc etc and outliner) on 3 different computers.

The Vim 8.0 with patches 1-1568 supplied with openSUSE 15.3 is nearly functional with git install of vim outliner.
(The following behavior occurs only with .otl files.   and with vim 8.2.2825).

Side note: I do not find that Vim 8.2 is compatible AT ALL with install locations from prior versions of vim outliner.
I have only found the VIM 8.2 plugin location change documented in recent months.

 =================================
THE PROBLEM I EXPERIENCED:
The y (yank) and p (paste) behave very oddly.  Paste (p) puts a yanked line (yy)
in the wrong place and at a different indent level.

1) Outline example.
    [_] line one
    [_] line two
           > sentence 1
           > sentence 2
           > sentence 3
    [_] line three
    [_] line four
2) Next item on list

If I 'zc' line two, hit 'dd' (delete) and hit 'p' to paste deletion with curson on line "1) Outline like this."
Normally it would pasted on the following line ... and just above "[_] line one " without changing the indent level.
Vim does correctly paste the lines in a file without the .otl extension.

Instead, using vim outliner (from git)  I get this outline:
 
1) Outline example.
    [_] line one
    [_] line three
    [_] line four
[_] line two
       > sentence 1
       > sentence 2
       > sentence 3
2) Next item on list

"[_] line two" and its indented members are promoted to the indent level where p is invoked and moved down and inserted
just above "2) Next item on listto the bottom of the paste level.

In my files with many lines at the same level ... the 'p' operation often deposits 
my deletion 400 or 500 lines down from the target line and at the promoted indent level. 

Disconcerting enough to make me forget what I was doing ... especially after a instinctive :w.

There is another paste operation deviation that ends up inserting the deleted (or yanked  yy) contents in the middle
of another line.   I have not figured out yet the triggering keystrokes. 

It happens infrequently and my mind is on autopilot and I have not figured out the triggering keystrokes.  
But the behavior is not as expected.

I keep hoping a future Vim release will clean up this change.   
It has been a year now ... still getting better but still a problem.

So, from the fellow beyond being saved
  .... does anybody else notice this problem?

So far I have not found another person I can confirm using vim outliner with vim 8.2.

Mouse driven world where everyone has shifted entirely to using an IDE editor and wordprocessor?

Thank you,
laternser
